### PR TITLE
chore: webui react storybook for StateSelectFilter component [DET-4039]

### PR DESCRIPTION
**Pull request by apizzini**
## Description

This PR addresses the request of DET-4039 JIRA issue.
Note that storybooks are still not working on master due to serverAddress issue (addressed in this PR: https://github.com/determined-ai/determined/pull/1240/commits).

## Test Plan

Storybook working fine.

## Commentary (optional)

Screenshots:
Schermata 2020-09-04 alle 10 04 30
Schermata 2020-09-04 alle 10 04 36

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

### CODE REVIEW

**Review by hkang1 — CHANGES_REQUESTED** [review]
I just realized that `onChanges` should not be required on the `StateSelectfilter` based on the stories you wrote. Would you mind updating `StateSelectFilter` to make `onChange` prop optional so we don't need to pass in a blank function? 🙏

**Comment by apizzini** [comment]
@hkang1 changed `StateSelectfilter` to make `onChanges ` optional.

File: webui/react/src/components/StateSelectFilter.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import StateSelectFilter from './StateSelectFilter';
+
+export default {
+  component: StateSelectFilter,
+  decorators: [ withKnobs ],
+  title: 'StateSelectFilter',
+};
+
+export const Default = (): React.ReactNode => (
+  <StateSelectFilter />
```
Comment (hkang1) [inline webui/react/src/components/StateSelectFilter.stories.tsx:13]: Thinking the same task as the `UserSelectFilter` of adding `onChange` event to this story, just to test and preserve the state of the selected value (user).

**Review by hkang1 — APPROVED** [review]
LGTM! 🎉

**Pull request merged**